compiler-core/src/build/module_loader.rs

```diff
@@ -20,6 +22,35 @@ use crate::{
     Error, Result,
 };
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
+pub(crate) struct SourceDigest(Output<Sha256>);
+
+impl SourceDigest {
+    pub(crate) fn new(source: &str) -> Self {
+        let mut hasher = Sha256::new();
+        hasher.update(source.as_bytes());
+        SourceDigest(hasher.finalize())
+    }
+}
+
+impl TryFrom<String> for SourceDigest {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let raw_digest = base16::decode(value.as_bytes()).map_err(|e| e.to_string())?;
```

Comment:
This is stored in a binary file so no need to convert to a string like this, we can just store the raw bytes of the fingerprint. Avoid some extra work.

Reply:
This is no longer needed, since xxhash outputs integers, which are serializable.